### DISCUSSION TITLE
Matching aromatic compounds with kekulized SMARTS

### DISCUSSION THREAD

**Discussion by jasondbiggs** [rdkit/rdkit#4413] [Development]
Say I have some SMARTS pattern picked up from some list of functional groups, or some list of reactions, and it might be tailored to a kekulized version of a molecule.  Is there any way to match it to a molecule?

```
>>> m = MolFromSmiles('c1ccccc1')
>>> rdmolops.Kekulize(m)
>>> patt = MolFromSmarts('[C]1[CH]=[CH][CH]=[CH][CH]=1')
>>> m.GetSubstructMatch(patt)
()
>>> ps = Chem.SubstructMatchParameters()
>>> ps.aromaticMatchesConjugated = True
>>> m.GetSubstructMatch(patt, ps)
()
```
Is there any way to 'fix' that SMARTS string?

**Comment by greglandrum** [comment]
@jasondbiggs, the SMARTS string is fine. The problem is that the molecule still has aromatic flags on it even though the bonds have been kekulized. This is easily fixable:
```
In [39]: patt1 = Chem.MolFromSmarts('[C]1[CH]=[CH][CH]=[CH][CH]=1')

In [40]: patt2 = Chem.MolFromSmarts('c1ccccc1')

In [41]: m = Chem.MolFromSmiles('c1ccccc1')

In [42]: km = Chem.Mol(m)

In [43]: Chem.Kekulize(km,clearAromaticFlags=True)

In [44]: [km.HasSubstructMatch(p) for p in (patt1,patt2)]
Out[44]: [True, False]
```

As an aside: the `aromaticMatchesConjugated` parameter only affects results when neither bond being compared has a query associated with it. It's not working here because bonds from SMARTS *always* have queries.
